registration-processor/pre-processor/registration-processor-cmd-validator-stage/src/main/java/io/mosip/registration/processor/stages/cmdvalidator/DeviceValidator.java

```diff
@@ -62,81 +87,148 @@
+					StatusUtil.DEVICE_VALIDATION_FAILED.getCode(),StatusUtil.DEVICE_VALIDATION_FAILED.getMessage());
+		}
+		
+		}
+		}else {
+			throw new BaseCheckedException(
+					StatusUtil.DEVICE_VALIDATION_FAILED.getCode(),StatusUtil.DEVICE_VALIDATION_FAILED.getMessage()+"-->Others info is not prsent in packet");
 		}
 	}
 
-	/**
-	 * Converts new registered device to masterdata registered device
-	 *
-	 * @param deviceDetails
-	 * @return
-	 */
-	private RegisteredDevice convert(NewRegisteredDevice deviceDetails) {
-		DigitalId digitalId = new DigitalId();
-		NewDigitalId newDigitalId = deviceDetails.getDigitalId();
-		if (newDigitalId != null) {
-			digitalId.setDateTime(newDigitalId.getDateTime());
-			digitalId.setDeviceSubType(newDigitalId.getDeviceSubType());
-			digitalId.setDp(newDigitalId.getDeviceProvider());
-			digitalId.setDpId(newDigitalId.getDeviceProviderId());
-			digitalId.setMake(newDigitalId.getMake());
-			digitalId.setModel(newDigitalId.getModel());
-			digitalId.setSerialNo(newDigitalId.getSerialNo());
-			digitalId.setType(newDigitalId.getType());
+	private boolean isDeviceHotlisted(String deviceCode, String payloadTimestamp) throws JsonParseException, JsonMappingException, JsonProcessingException, IOException, JSONException, BaseCheckedException {
+		List<String> pathSegments=new ArrayList<>();
+		pathSegments.add("DEVICE");
+		pathSegments.add(deviceCode);
+		ResponseWrapper<?> responseWrapper = (ResponseWrapper<?>) registrationProcessorRestService
+				.getApi(ApiName.HOTLIST, pathSegments,"", "", ResponseWrapper.class);
+		if(responseWrapper.getResponse() !=null) {
+			HotlistRequestResponseDTO hotListResponse=mapper.readValue(mapper.writeValueAsString(responseWrapper.getResponse()),
+					HotlistRequestResponseDTO.class);
+		DateTimeFormatter format = DateTimeFormatter.ofPattern(env.getProperty(DATETIME_PATTERN));
+		
+		LocalDateTime payloadTime = LocalDateTime.parse(payloadTimestamp, format);
+		if(hotListResponse.getExpiryTimestamp()!=null) {
+		
+		if(hotListResponse.getStatus().equalsIgnoreCase("BLOCKED") &&
+				payloadTime.isBefore(hotListResponse.getExpiryTimestamp())) {
+			throw new BaseCheckedException(
+					StatusUtil.DEVICE_HOTLISTED.getCode(),
+					StatusUtil.DEVICE_HOTLISTED.getMessage());
+		}
+		}
+		else {
+			if(hotListResponse.getStatus().equalsIgnoreCase("BLOCKED")) {
+				throw new BaseCheckedException(
+						StatusUtil.DEVICE_HOTLISTED.getCode(),
+						StatusUtil.DEVICE_HOTLISTED.getMessage());
+			}
 		}
-		RegisteredDevice registeredDevice = new RegisteredDevice(deviceDetails.getDeviceCode(),
-				deviceDetails.getDeviceServiceVersion(), digitalId);
-		return registeredDevice;
+		 return true;
+		}
+		else {
+			throw new BaseCheckedException(
+					responseWrapper.getErrors().get(0).getErrorCode(),
+					responseWrapper.getErrors().get(0).getMessage());
+		}
+	}
+
+	private boolean validateTimeStamp(JSONObject payload, String packetCreationDate) throws JSONException, BaseCheckedException {
```

Comment:
Rename this method to validateDigitalIdTimestamp or validatePayloadTimestamp based on the requirement

Reply:
Renamed to validateDigitalIdTimestamp .